src/palimpzest/core/data/datasources.py

```diff
@@ -41,8 +41,12 @@ class AbstractDataSource(abc.ABC):
     In the future, programmers can implement their own DataSources using custom Schemas.
     """
 
-    def __init__(self, schema: Schema) -> None:
-        self._schema = schema
+    def __init__(self, schema: type[Schema] | list[dict]) -> None:
```

Comment:
Should we continue to support Schema?

Reply:
I would like to, because I think some power users in the future may agree with our initial gut feeling that a class-based approach to defining fields & schemas was elegant, and there's little harm in supporting both